lib/internal/process/worker_thread_only.js

```diff
@@ -2,32 +2,25 @@
 
 // This file contains process bootstrappers that can only be
 // run in the worker thread.
-const {
-  getEnvMessagePort
-} = internalBinding('worker');
 
 const {
-  kWaitingStreams,
-  ReadableWorkerStdio,
-  WritableWorkerStdio
+  getWorkerStdio
 } = require('internal/worker/io');
 
 const {
   codes: { ERR_WORKER_UNSUPPORTED_OPERATION }
 } = require('internal/errors');
-const workerStdio = {};
 
+let workerStdio;
+function lazyWorkerStdio() {
+  if (!workerStdio) workerStdio = getWorkerStdio();
+  return workerStdio;
+}
 function initializeWorkerStdio() {
-  const port = getEnvMessagePort();
-  port[kWaitingStreams] = 0;
-  workerStdio.stdin = new ReadableWorkerStdio(port, 'stdin');
-  workerStdio.stdout = new WritableWorkerStdio(port, 'stdout');
-  workerStdio.stderr = new WritableWorkerStdio(port, 'stderr');
-
   return {
-    getStdout() { return workerStdio.stdout; },
-    getStderr() { return workerStdio.stderr; },
-    getStdin() { return workerStdio.stdin; }
+    getStdout() { return lazyWorkerStdio().stdout; },
+    getStderr() { return lazyWorkerStdio().stderr; },
+    getStdin() { return lazyWorkerStdio().stdin; }
```

Comment:
Can we just call `initializeWorkerStdio()` later? I don’t think this is specific to worker threads, we probably don’t want to create handles for stdio in the main thread early either…

Reply:
For now, we create the getters eagerly to define the process.stdin/stdout/stderr, but we do not actually access them during bootstrap. The handles are only created the first time these streams are accessed, for workers, it is usually done eagerly when executing the main script, for the main thread, it's probably done when the users actually access any of these streams (only until then would we know if those should be backed by file streams, ttys, or sockets etc)

Reply:
If we don’t allow accessing stdio before bootstrap is complete, then it makes sense to just initialize stdio later than that, right? I’m okay with this as it is, I just want to understand why we’re making this lazy as opposed to creating these before running the main script

Reply:
We could attach the stdios eagerly after bootstrap, yes, but the getters need to be available by the time the console methods are accessed, so we need to be careful not to do that later than the first console method call - it's also a bit hard to tell when the console methods would be called for the first time in the case of `--inspect-brk` or `--inspect-brk-node`. so might as well just setup the getters early.